references/depth/stereo/cascade_evaluation.py

```diff
@@ -182,6 +157,23 @@ def preprocessing(image_left, image_right, disp, valid_disp_mask):
         
         metrics[name] = {}
         val_dataset = make_dataset(name, args.dataset_root, transforms=preprocessing)
+        
+        effective_batch_size = args.batch_size * args.world_size
+        can_do_distributed_test = effective_batch_size < len(val_dataset)
+        
+        if args.distributed and can_do_distributed_test:
+            sampler = torch.utils.data.distributed.DistributedSampler(val_dataset, shuffle=False, drop_last=False)
+        else:
+            sampler = torch.utils.data.SequentialSampler(val_dataset)
+        
+        val_loader = torch.utils.data.DataLoader(
```

Comment:
NIT, Similar with my comment in train.py, we should consider moving val_loader creation to main

Reply:
@YosuaMichael it could be moved, only observation is that it should be converted to a dictionary of loaders in order to still know which datasets is being handled during evaluation in order to correctly log it